packages/react-tinacms-editor/src/state/plugins/links/index.tsx

```diff
@@ -54,15 +55,26 @@ export function links(): Plugin {
           }
         }
 
+        const { editorFocused } = commonPluginKey.getState(state)
+        const { anchor, head } = state.selection
+        let pos = anchor < head ? anchor : head
+        pos = state.selection.empty ? pos : pos + 1
+
         if (
-          state.selection.$anchor
+          state.doc
+            .resolve(pos)
             .marks()
             .some((mark: Mark) => mark.type === state.schema.marks.link)
```

Comment:
Seeing as this logic is shared between here and `Menu`, maybe it should be in a helper funciton. Maybe something like `checkForMark(mark: MarkType)`. I'm only slightly worried about removing the duplication. I'm more concerned with writing some tests explaining edge cases that it's trying to catch.

Reply:
Make sense, I will do this cleanup 👍

Reply:
Hey @ncphillips : PR is updated.